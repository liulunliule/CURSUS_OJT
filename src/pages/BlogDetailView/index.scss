.blog-detail {
  padding: 20px;
  max-width: 830px;
  margin: 0 auto;

  .blog-title {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 20px;
    margin-top: 85px;

    h1 {
      font-size: 2em;
      margin: 0;
    }

    .back-to-blog {
      background-color: red;
      color: #fff;
      padding: 10px 20px;
      text-decoration: none;
      border-radius: 5px;
      transition: background-color 0.3s;

      &:hover {
        background-color: black;
      }
    }
  }

  .blog-content {
    img {
      width: 100%;
      height: auto;
      margin-bottom: 20px;
    }

    .blog-meta {
      font-size: 0.9em;
      color: #888;
      margin-bottom: 20px;
    }

    p {
      font-size: 1em;
      line-height: 1.6;
      margin-bottom: 20px;
    }

    h2 {
      font-size: 1.4em;
      margin: 20px 0 10px;
    }
  }

  .pagination {
    display: flex;
    justify-content: space-between;
    margin: 20px 0;

    a {
      font-size: 0.9em;
      color: #007bff;
      text-decoration: none;
      transition: color 0.3s;

      &:hover {
        color: #0056b3;
      }
    }
  }
}