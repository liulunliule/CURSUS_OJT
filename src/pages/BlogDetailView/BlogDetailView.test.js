import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom';
import BlogDetailView from '.';

test('renders Blog Detail View component correctly', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <BlogDetailView />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});