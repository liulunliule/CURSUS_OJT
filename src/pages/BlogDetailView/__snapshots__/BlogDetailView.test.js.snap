// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Blog Detail View component correctly 1`] = `
<DocumentFragment>
  <div
    class="blog-detail"
  >
    <div
      class="blog-title"
    >
      <h1>
        Blog Title Here
      </h1>
      <a
        class="back-to-blog"
        href="/secondLayout/blog"
      >
        Back to Blog
      </a>
    </div>
    <div
      class="blog-content"
    >
      <img
        alt="Blog detail"
        src="blog-detail-image.jpg"
      />
      <div
        class="blog-meta"
      >
        <span>
          109k views
        </span>
         • 
        <span>
          March 10, 2020
        </span>
      </div>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur volutpat maximus pellentesque. Integer sem enim, luctus at nibh at, condimentum sagittis sapien. Sed tempus ipsum erat, sit amet efficitur velit interdum eu. Vestibulum hendrerit id dolor eu scelerisque. Phasellus ex dui, consequat nec feugiat eu, dapibus eget ante. Sed sodales interdum dui, at euismod mi feugiat hendrerit. Suspendisse auctor libero in tempor mollis. Nulla et dolor velit. Aliquam sit amet luctus quam.
      </p>
      <h2>
        Why did you decide to teach on Cursus?
      </h2>
      <p>
        Nam a egestas libero, eget eleifend turpis. Sed id ipsum a ipsum aliquam laoreet sit amet sit amet nibh. Proin dapibus, libero sed posuere rhoncus, orci mi cursus enim, at accumsan eros massa lacinia mi. Nunc eget finibus felis, volutpat malesuada sem. Aliquam ac nisl pellentesque, varius neque sit amet, porttitor nunc. Nullam elit tellus, dapibus non eleifend sed, hendrerit eget velit. Aliquam ut felis dictum, tincidunt magna vitae, aliquam massa. In porttitor tristique quam, non dignissim sapien pharetra ultrices. Cras non ante non velit mollis mollis. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque et bibendum urna, eget consequat sapien. Integer sed condimentum nibh. Integer id neque tristique, lobortis massa ac, dapibus nibh. Donec nulla odio, porttitor ac rutrum eget, volutpat a velit. Curabitur et enim quis diam congue dictum et vitae dui. Nulla tortor orci, luctus a pretium vel, ultrices porta nisl.
      </p>
      <p>
        Etiam lobortis dictum tincidunt. Interdum et malesuada fames ac ante ipsum primis in faucibus. Suspendisse ultricies efficitur dui, suscipit tempus elit condimentum quis. Duis sed vestibulum tortor, eget cursus odio.
      </p>
      <h2>
        Did you have any prior on- or offline teaching experience prior to publishing your first Cursus course?
      </h2>
      <p>
        Morbi lectus nunc, lacinia ut consequat a, semper vel erat. Duis ut lacus nec dui sodales mattis. Mauris tellus dolor, pulvinar sit amet pretium a, malesuada sed tellus. Aliquam ultrices elit neque, quis lacinia ex porttitor non. Donec ac iaculis turpis. Nulla lacinia enim quis orci aliquam, non cursus urna pellentesque. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Phasellus in mi a nisi auctor interdum. Vivamus faucibus magna sed elit interdum consequat. Vestibulum eu tortor vel ante feugiat faucibus quis et urna. Quisque interdum ac enim eu tempus. Fusce viverra, lectus egestas tincidunt cursus, tortor sapien convallis metus, vitae auctor risus metus vel nisi. Aenean dapibus bibendum mauris ut iaculis.
      </p>
      <h2>
        What are the most rewarding aspects of teaching on Cursus?
      </h2>
      <p>
        Quisque et bibendum urna, eget consequat sapien. Integer sed condimentum nibh. Integer id neque tristique, lobortis massa ac, dapibus nibh. Donec nulla odio, porttitor ac rutrum eget, volutpat a velit. Curabitur et enim quis diam congue dictum et vitae dui. Nulla tortor orci, luctus a pretium vel, ultrices porta nisl.
      </p>
    </div>
    <div
      class="pagination"
    >
      <a
        class="previous"
        href="#"
      >
        « Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </a>
      <a
        class="next"
        href="#"
      >
        Vestibulum vulputate nulla quis dignissim ultricies. »
      </a>
    </div>
  </div>
</DocumentFragment>
`;